roles/initramfs/tasks/main.yml

```diff
@@ -0,0 +1,38 @@
+---
+- name: Make sure the required related facts are available
+  ansible.builtin.setup:
+    gather_subset:
+    - "!all"
+    - "!min"
+    - kernel
+
+- name: Get kernel version
+  ansible.builtin.set_fact:
+    _kernel_version: "{{ ansible_facts.kernel }}"
+
+- name: Create a backup of the current initramfs
+  ansible.builtin.copy:
+    remote_src: true
+    src: /boot/initramfs-{{ _kernel_version }}.img
+    dest: /boot/initramfs-{{ _kernel_version }}.img.{{ initramfs_backup_extension }}
+    mode: "0600"
+
+- name: Create a new initramfs with the extend boot module
+  ansible.builtin.command: '/usr/sbin/dracut {{ ((initramfs_add_modules | length) > 0) | ternary("-a", "") }} "{{ initramfs_add_modules }}" --kver {{ _kernel_version }} --force'
+  changed_when: true
+
+- name: Reboot the server
```

Comment:
Why does this role have to reboot the server? Wouldn't it make sense for that to be done by the caller task? Something like this:
- initram role: create_initramfs task
- caller role: reboot task
- initram role: clear_initramfs task

Reply:
If every caller keeps this flow, then I don't see the point in not having it in the initramfs role. The reason I would see to split it is if the caller will have something to do between the creation and the reboot or between the reboot and the cleanup. Currently, I don't see that. So, why not have the initramfs role handle it?